flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/Observer.java

```diff
@@ -127,14 +136,26 @@ private void observeJmDeployment(
                     return;
                 }
             }
+            // checking the pod is expensive; we only do it when the deployment isn't ready
+            PodList jmPods = flinkService.getJmPodList(flinkApp, effectiveConfig);
+            for (Pod pod : jmPods.getItems()) {
+                for (ContainerStatus cs : pod.getStatus().getContainerStatuses()) {
+                    ContainerStateWaiting csw = cs.getState().getWaiting();
+                    if ("CrashLoopBackOff".equals(csw.getReason())) {
```

Comment:
Could this check be `!StringUtils.isEmpty(csw.getReason())  && csw.getReason().toLowerCase().contains("CrashLoopBackOff".toLowerCase())`?